### ISSUE TITLE
Feedback-driven distillation: fine-tune local model from usage data

### ISSUE THREAD

**Issue by ryanstpierre** [Threshold-Labs/deckydecks#5]
## Summary

Use accumulated feedback + session data to fine-tune a smaller model for deck generation, eventually replacing API calls entirely.

## The flywheel

1. Sonnet generates deck → user reviews → feedback refines it
2. (original deck, feedback, refined deck) = training pair
3. LoRA fine-tune Qwen 14-32B on Spark
4. Fine-tuned model generates → user reviews → better training data → retrain

## Training data sources

- **Completed sessions** (high completion rate = good deck structure)
- **Accepted feedback** (user validated the refinement)
- **Dwell times** (long engaged time on a node = good content)
- **Dropout points** (nodes where users leave = bad content)

## Export endpoint

`GET /api/training-export` — joins D1 sessions + feedback with R2 deck JSON, outputs JSONL training pairs:
```jsonl
{"input": {"source": "...", "context": {...}}, "output": {"deck": {...}}, "quality_signals": {"completion_rate": 0.85, "avg_dwell": 12.3}}
```

## JIT node patching (incremental generation)

Before full distillation, implement surgical fixes:
- When a node has high dropout, auto-regenerate just that node
- When feedback says "confusing", rewrite that single node
- Patch = ~200-500 tokens, not a full 3-4K deck regen
- Small enough for a 7B model to handle reliably

## Timeline

| Phase | Examples needed | What you get |
|-------|----------------|-------------|
| ~30 good decks | 1-2 weeks of use | Basic LoRA, simple decks work |
| ~100 decks + feedback pairs | 1-2 months | Reliable branching, good quality |
| Ongoing | Continuous | Model keeps improving |

## Depends on

- #3 (deployed with D1 for session/feedback storage)
- #4 (Spark vLLM for serving fine-tuned model)
